### PR TITLE
Use boto3

### PR DESCRIPTION
**Pull request by hanpeter**
## What does this PR do?
This PR is a major refactoring of this branch to make it use Boto3

## Why is this change being made?
Boto3 should replace Boto2.

## Where should the reviewer start?
`krux_ec2/ec2.py`

## How was this tested? How can the reviewer verify your testing?
This was tested in the development environment and through unit tests.

## Completion checklist

- [x] The pull request has been appropriately labelled according to
  our [conventions](https://kruxdigital.jira.com/wiki/display/EN/Changes+and+Peer+Review)
- [x] Dependencies have been documented, deployed, and verified as
      appropriate.
- [x] Use `krux.object.Object`.
- [x] Decommission use of `client`s and replace the `waiter`s with `resource`'s built in [waiting functions](https://boto3.readthedocs.io/en/stable/reference/services/ec2.html#EC2.Instance.wait_until_exists)
- [x] Allow `*args` and `**kwargs` in methods so that the extra arguments can be passed to boto functions.
- [x] The change has unit & integration tests as appropriate.
- [x] Readme has been updated.
- [x] Complete docstring update.
- [x] Stakeholders have been notified. Workflow-impacting changes have
      been appropriately socialized to avoid surprises.

### CODE REVIEW

File: krux_ec2/ec2.py [hunk 1]
```diff
@@ -114,141 +119,192 @@ def add_ec2_cli_arguments(parser, include_boto_arguments=True):
...
+    # OK, so, on larger instances, extra devices only show up if you tell them to associate with a block device.
+    # These EBS AMIs don't set this up, so we have to. sdb will always be ephemeral0,
+    # which is how we've always done it. If there are more devices, they will get sdc,sdd,sde.
+    # NOTE: see mounts.pp in kbase for how-we-deal-with-these.
+    _BLOCK_DEVICE_MAP = [{
+        'VirtualName': 'ephemeral0',
+        'DeviceName': '/dev/sdb',
+    }, {
+        'VirtualName': 'ephemeral1',
+        'DeviceName': '/dev/sdc',
```
Comment (kruxErinWillingham) [inline krux_ec2/ec2.py]: We should make sure to test / review this section well. I vaguely recall different devices potentially showing up as different names w/ AWS. Maybe on the VPC side of things? I can't remember any specifics but let's put some time here to make sure the different servers stand out how we want them to.

Comment (hanpeter) [inline krux_ec2/ec2.py]: @kruxErinWillingham Can you suggest some ways to test this section? Should I just spin up multiple nodes with various different instance types?

Comment (paul-krohn) [inline krux_ec2/ec2.py]: I think what Erin is referring to is that on some distributions (at least CentOS), `sdb` becomes `/dev/xvdg`. You read that right, off-by-four.

The API, as far as I know, doesn't provide a map of instance type to ephemeral drives, so we are left to maintain this on our own; there are instances with up to 24 ephemeral devices. It appears to be harmless to set up "extra" devices, 4 covers a large percentage of our use cases, but might be maddening when it does not, and setting up 24 feels ... more wrong? :(

Comment (hanpeter) [inline krux_ec2/ec2.py]: Well, the main use case for this repo is to support Krux, but I see the point. A future work should be exposing the device mounting with the default as what is coded here. That way, you can override it if you wish.

I have run this with `c3.large`, `m3.xlarge`, `m3.2xlarge`, `c3.4xlarge`, `c3.xlarge`, `d2.2xlarge`, `i2.2xlarge`, `c3.2xlarge`, `c1.xlarge`, `m1.small`. (These are top 10 instance types we use.) Bootstrap succeeded in all instances. I will consider this as good enough for now.

**Pull request merged**